### ISSUE TITLE
[help] how to fix home-manager reverting the setting after reboot

### ISSUE THREAD

**Issue by nakoo** [itsmeyashraj/dotfiles#1]
Hello, I came across your Reddit post about home-manager issue that reverts settings after reboot. I am experiencing a similar problem and was hoping you could assist me with it. Could you please share the solution you used to fix the issue? I have already checked your source code, but I'm still unsure about the root cause of the issue.

https://www.reddit.com/r/NixOS/comments/112ekgm/homemanager_switch_does_not_retain_changes_after/

**Comment by mobsenpai** [comment]
I use nixos-rebuild for creating os changes and home-manager switch to check the home configuration without having to rebuild the whole system. I don't really know if it's intended to not retain changes but I think its useful if you have to just check your new configs like window managers config or other non system configs. There was a comment that we should use nixos-rebuild for system changes and home-manager is for when we are not on natively nixos and using nix package manager on other os. 

In short:
Use home-manager : to test non system changes (will revert after restart), on non nixos system where nixos-rebuild is not available
Use nixos-rebuild : for whole system changes (home and system both)

**Comment by nakoo** [comment]
> In short: Use home-manager : to test non system changes (will revert after restart), on non nixos system where nixos-rebuild is not available Use nixos-rebuild : for whole system changes (home and system both)

Thank you! home-manager manual confused me a lot 😅
